### ISSUE TITLE
[REVIEW]: hmds: An R Package for Heuristic High and Multi Dimensional Scaling

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#214]
**Submitting author:** @jirotubuyaki (<a href="http://orcid.org/0000-0001-6871-7727">Masashi Okada</a>)
**Repository:** <a href="https://github.com/jirotubuyaki/hmds" target ="_blank">https://github.com/jirotubuyaki/hmds</a>
**Version:** version 1.0
**Editor:** @karthik
**Reviewer:** @kevinushey
**Archive:** Pending

## Status

[![status](http://joss.theoj.org/papers/bebd059342063463594922799cd98b35/status.svg)](http://joss.theoj.org/papers/bebd059342063463594922799cd98b35)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/bebd059342063463594922799cd98b35"><img src="http://joss.theoj.org/papers/bebd059342063463594922799cd98b35/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/bebd059342063463594922799cd98b35/status.svg)](http://joss.theoj.org/papers/bebd059342063463594922799cd98b35)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/jirotubuyaki/hmds" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jirotubuyaki/hmds">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (version 1.0)?
- [x] **Authorship:** Has the submitting author (@jirotubuyaki) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @kevinushey it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by arfon** [comment]
:wave: @kevinushey, how are you getting on with your review?

**Comment by kevinushey** [comment]
Sorry for the delay! I'm hoping to have this done by next weekend.

**Comment by kevinushey** [comment]
I've finished my initial review for now -- there's a number of things that still need to be finished before a full review can be conducted; that is, I don't think the paper is ready yet. Some comments:

>  Functionality: Have the functional claims of the software been confirmed?

Currently, the package / paper does not outline other competing methods for multidimensional scaling (in R or otherwise). A discussion of how this package compares to other approaches to dimensional reduction would be warranted. For example, what solutions already exist for multidimensional scaling in R, and what particular niche does this package fill? I'm not as familiar with the landscape here but a discussion of these in the paper + a comparison in the package vignettes etc. would be helpful.

>  A statement of need: Do the authors clearly state what problems the software is designed to solve and who the target audience is?

Somewhat related to the above -- what are the competing solutions already available, and what niche does this package fill exactly?

>  Example usage: Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).

Currently, I only see one example usage in the package -- this could be expanded.

>  Automated tests: Are there automated tests or manual steps described so that the function of the software can be verified?

There are currently no automated tests -- some test should be added to ensure that the software behaves well with unexpected inputs.

>  Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

Currently, contributions are currently requested through e-mail; it would likely be more productive to accept contributions through (for example) GitHub issues, and supply a `CONTRIBUTING.md` to indicate what's expected from contributions.

**Comment by karthik** [comment]
@jirotubuyaki  Do you need more time to respond to these comments? Can you please update us when you get a chance?

**Comment by karthik** [comment]
@jirotubuyaki Just a friendly reminder to give us an update when you are know more.

**Comment by jirotubuyaki** [comment]
I am sorry for being late to improve my software.
I plan some improvement for my GitHub repository.

**Comment by arfon** [comment]
Hi @jirotubuyaki - how are you getting along here? Are you still interested in pursuing this publication in JOSS?

**Comment by jirotubuyaki** [comment]
I would like to give up for submit my paper.
I must learn many things for writing papers.
Thank you very much.

**Comment by arfon** [comment]
@kevinushey & @karthik - thanks for your help with this submission. Unfortunately, it looks like this one isn't going to make it through to publication.
